src/org/ods/component/RolloutOpenShiftDeploymentStage.groovy

```diff
@@ -108,52 +111,62 @@ class RolloutOpenShiftDeploymentStage extends Stage {
+            && !isHelmDeployment) {
+            steps.error "Deployment resources cannot be used in a NON HELM orchestration pipeline."
             return
         }
         def originalDeploymentVersions = fetchOriginalVersions(deploymentResources)
 
         def refreshResources = false
         def paused = true
         try {
-            openShift.bulkPause(context.targetProject, deploymentResources)
+            if (!isHelmDeployment) {
+                openShift.bulkPause(context.targetProject, deploymentResources)
+            }
 
             // Tag images which have been built in this pipeline from cd project into target project
             retagImages(context.targetProject, getBuiltImages())
 
-            if (steps.fileExists("${options.chartDir}/Chart.yaml")) {
-                if (context.triggeredByOrchestrationPipeline) {
-                    steps.error "Helm cannot be used in the orchestration pipeline yet."
-                    return
-                }
+            if (isHelmDeployment) {
+                logger.info "Rolling out ${context.componentId} with HELM, selector: ${options.selector}"
                 refreshResources = true
                 helmUpgrade(context.targetProject)
-            } else if (steps.fileExists(options.openshiftDir)) {
+            }
+            if (steps.fileExists(options.openshiftDir)) {
                 refreshResources = true
                 tailorApply(context.targetProject)
             }
+
             if (refreshResources) {
                 deploymentResources = openShift.getResourcesForComponent(
                     context.targetProject, DEPLOYMENT_KINDS, options.selector
                 )
             }
 
-            def metadata = new OpenShiftResourceMetadata(
-                steps,
-                context.properties,
-                options.properties,
-                logger,
-                openShift
-            )
-            metadata.updateMetadata(true, deploymentResources)
-
-            def rolloutData = rollout(deploymentResources, originalDeploymentVersions)
-            paused = false
-            return rolloutData
+            def rolloutData = [:]
+            if (!isHelmDeployment) {
+                def metadata = new OpenShiftResourceMetadata(
+                    steps,
+                    context.properties,
+                    options.properties,
+                    logger,
+                    openShift
+                )
+                metadata.updateMetadata(true, deploymentResources)
+                rolloutData = rollout(deploymentResources, originalDeploymentVersions)
+                paused = false
+                return rolloutData
+            } else {
+                rolloutData = rollout(deploymentResources, originalDeploymentVersions, true)
+                paused = false
+                return rolloutData
+            }
         } finally {
-            if (paused) {
+            if (paused && !isHelmDeployment) {
```

Comment:
Please put Tailor and Helm-related logic into a Strategy class and apply the one or the other strategy vs. having a single logic that tries to handle both. This is otherwise hard to read and mock/test.

Example: class Tailor {
    prepare(context,...)
    apply()
    isApplicable()
    ...
}